### PR TITLE
[15.0][MIG] hr_timesheet_sheet_attendance

### PR DESCRIPTION
**Pull request by JasminSForgeFlow**
Standard Migration

@ForgeFlow

### CODE REVIEW

File: hr_timesheet_sheet_attendance/readme/USAGE.rst [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+* Goto tab Attendances on timesheet form
+    - You can see there your current status checkin/checkout
+    - You also can create attendance by clicking on button Check In/Check Out on right side
+    - You can see your attendance that belongs to current timesheet on left side in same tab
+* 'Total Attendance' is total working time based on your attendance
+* 'Difference' is the difference betwwen total attandance time and working time (sum(attendace-time) - sum(unit amount in timessheet lines))
+* Two smart buttons are present on top-right corner of timesheet form
+    - First one(with time icon) will take you list of your timesheets (by default filter timesheets related to current timesheet-sheet)
+    - Second one(labeled as Attendances) will take you to list of your attendances (by default filter ateendances related to current timesheet-sheet)
+* It prevents to change in any attendance ralted to timesheet-sheet that already has submitted
```
Comment (AaronHForgeFlow) [inline hr_timesheet_sheet_attendance/readme/USAGE.rst]:
```suggestion
* It prevents to change in any attendance related to timesheet-sheet that already has submitted
```

File: hr_timesheet_sheet_attendance/readme/INSTALL.rst [hunk 2]
```diff
@@ -0,0 +1,4 @@
+This module relies on:
+
+* The OCA module 'HR Timesheet Sheet', and can be downloaded from
+  Github: https://github.com/OCA/hr-timesheet/tree/12.0/hr_timesheet_sheet
```
Comment (AaronHForgeFlow) [inline hr_timesheet_sheet_attendance/readme/INSTALL.rst]:
```suggestion
  Github: https://github.com/OCA/hr-timesheet/tree/15.0/hr_timesheet_sheet
```

**Review by AaronHForgeFlow — COMMENTED** [review]
Some minor comments, They are probably not related to the migration itself, but should be fixed IMHO.

hr_timesheet_sheet_attendance_work_hours

File: hr_timesheet_sheet_attendance/models/hr_timesheet_sheet.py [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+        related="employee_id.attendance_state", string="Current Status"
+    )
+    attendance_count = fields.Integer(compute="_compute_attendance_count")
+
+    def attendance_action_change(self):
+        """Call attendance_action_change to
+        perform Check In/Check Out action
+        Returns last attendance record"""
+
+        return self.employee_id._attendance_action_change()
```
Comment (GuillemCForgeFlow) [inline hr_timesheet_sheet_attendance/models/hr_timesheet_sheet.py:68]: By having this button in the timesheet, there could perhaps be a case where the user does two clicks in a fast way, then what would happen is that the check-in and check-out times are the same. 

If this happens, the attendance gets blocked as there is an Odoo standard check which verifies that the attendance check-in time is strictly minor than the check-out time (https://github.com/odoo/odoo/blob/15.0/addons/hr_attendance/models/hr_attendance.py#L78).

I don't know how should we address this, or perhaps skip it as it is an unlikely case. Any thoughts on this?

File: hr_timesheet_sheet_attendance/readme/USAGE.rst (hunk 1 above)
Comment (JasminSForgeFlow) [reply hr_timesheet_sheet_attendance/readme/USAGE.rst]: It is updated,
Thanks

File: hr_timesheet_sheet_attendance/readme/INSTALL.rst (hunk 2 above)
Comment (JasminSForgeFlow) [reply hr_timesheet_sheet_attendance/readme/INSTALL.rst]: It is updated,
Thanks

File: hr_timesheet_sheet_attendance/models/hr_timesheet_sheet.py (hunk 3 above)
Comment (AaronHForgeFlow) [reply hr_timesheet_sheet_attendance/models/hr_timesheet_sheet.py:68]: @JasminSForgeFlow  was this comment attended somehow? Perhaps we can just add a note in the readme as known issues. WDYT?

**Review by AaronHForgeFlow — APPROVED** [review]
Functional review

File: hr_timesheet_sheet_attendance/models/hr_timesheet_sheet.py (hunk 3 above)
Comment (JasminSForgeFlow) [reply hr_timesheet_sheet_attendance/models/hr_timesheet_sheet.py:68]: @AaronHForgeFlow yes I think we can add it in readme note as known issue.
Let me add it.

Edit: in addition, I think it is more of a framework issue

Comment (AaronHForgeFlow) [reply hr_timesheet_sheet_attendance/models/hr_timesheet_sheet.py:68]: @GuillemCForgeFlow are you fine with this?

Comment (GuillemCForgeFlow) [reply hr_timesheet_sheet_attendance/models/hr_timesheet_sheet.py:68]: @JasminSForgeFlow @AaronHForgeFlow Yes, I agree, we can leave it as a known issue.

Comment (AaronHForgeFlow) [reply hr_timesheet_sheet_attendance/models/hr_timesheet_sheet.py:68]: Thank you :) Good for approval then? :pray:

**Review by GuillemCForgeFlow — APPROVED** [review]
LGTM 👍🏽

**Comment by AaronHForgeFlow** [comment]
@OCA/human-resources-maintainers  I think this is ready :)

**Review by Saran440 — APPROVED** [review]
Code Reviewed. No changed. :+1:

**Comment by Saran440** [comment]
@JasminSForgeFlow Can you rebase commit `[IMP] hr_timesheet_sheet_attendance: Add roadmap in README` to `[MIG] hr_timesheet_sheet_attendance: Migration to 15.0` please.

**Comment by JasminSForgeFlow** [comment]
> @JasminSForgeFlow Can you rebase commit `[IMP] hr_timesheet_sheet_attendance: Add roadmap in README` to `[MIG] hr_timesheet_sheet_attendance: Migration to 15.0` please.

Done, Thanks

File: hr_timesheet_sheet_attendance/models/hr_timesheet_sheet.py [hunk 4]
```diff
@@ -0,0 +1,105 @@
...
+    )
+    total_difference = fields.Float(
+        compute="_compute_attendance_time",
+        string="Difference",
+    )
+    attendances_ids = fields.One2many(
+        comodel_name="hr.attendance", inverse_name="sheet_id", string="Attendances"
+    )
+    attendance_state = fields.Selection(
+        related="employee_id.attendance_state", string="Current Status"
```
Comment (ortlam) [inline hr_timesheet_sheet_attendance/models/hr_timesheet_sheet.py]:
```suggestion
        related="employee_id.attendance_state", related_sudo=True, string="Current Status"
```
Without the sudo a user has no access to their attendance_state due to the `hr.employee.public` split, which leads to a missing field JavaScript error in this module for the use of `attendance_state` in another field's `invisible` attribute.

Comment (JasminSForgeFlow) [reply hr_timesheet_sheet_attendance/models/hr_timesheet_sheet.py]: Updated as per suggestion,
Thanks

**Comment by AaronHForgeFlow** [comment]
@Saran440 is there something else pending for the merge?

**Pull request merged**